sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -27,21 +27,45 @@ import org.apache.spark.sql.catalyst.util.{ArrayData, DateTimeUtils, MapData}
 import org.apache.spark.sql.types._
 
 private[sql] class JacksonGenerator(
-    schema: StructType,
+    childType: DataType,
+    rowSchema: StructType,
```

Comment:
We can loose the type of `rowSchema` from `StructType` to general `DataType`. Since only `StructType` and `MapType` are valid options, we can check this in `JacksonGenerator`'s constructor.

Reply:
That's said, for now `JacksonGenerator` supports to write out a struct or an array of struct. After revising, it can also support to write out map, array of map.

Once a `JacksonGenerator` is initialized with a `MapType`, it is not valid to call the API `write(row: InternalRow)`.

Reply:
thanks for review =). I will follow your suggestion to change it. 
But I think `JacksonGenerator` only support write out an arbitrary map, it doesn't support to write out an array of map yet. Should I need to fix it? I think that maybe an issue for supporting arbitrary array?
Should I need to do some check for API `write(row: InternalRow)` calling?